src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -456,14 +510,22 @@ public virtual void WriteDeltaDeletedEntry(object graph, ODataWriter writer, ODa
         /// </summary>
         /// <param name="graph">The object to be written.</param>
         /// <param name="writer">The <see cref="ODataDeltaWriter" /> to be used for writing.</param>
-        /// <param name="writeContext">The <see cref="ODataSerializerContext"/>.</param>
+        /// <param name="writeContext">The <see cref="ODataSerializerContext"/>.</param>        
         public virtual async Task WriteDeltaDeletedEntryAsync(object graph, ODataWriter writer, ODataSerializerContext writeContext)
         {
-            ODataDeletedResource deletedResource = GetDeletedResource(graph);
-            if (deletedResource != null)
+            ODataResourceSerializer serializer = SerializerProvider.GetEdmTypeSerializer(_elementType) as ODataResourceSerializer;
+            ResourceContext resourceContext = serializer.GetResourceContext(graph, writeContext);
+            SelectExpandNode selectExpandNode = serializer.CreateSelectExpandNode(resourceContext);
+
+            if (selectExpandNode != null)
             {
-                await writer.WriteStartAsync(deletedResource);
-                await writer.WriteEndAsync();
+                ODataDeletedResource deletedResource = GetDeletedResource(graph, resourceContext, serializer, selectExpandNode, writeContext.IsUntyped);
+
+                if (deletedResource != null)
+                {
+                    await writer.WriteStartAsync(deletedResource);
```

Comment:
Do we need to continue execution on the same thread that awaited? I wonder if we want to call `ConfigureAwait(false)` here